src/chess_move.rs

```diff
@@ -353,23 +355,75 @@ impl ChessMove {
             }
 
             // takes is complicated, because of e.p.
-            if !takes {
-                if board.piece_on(m.get_dest()).is_some() {
-                    continue;
-                }
+            if !takes && board.piece_on(m.get_dest()).is_some() {
+                continue;
             }
 
-            if !ep && takes {
-                if board.piece_on(m.get_dest()).is_none() {
-                    continue;
-                }
+            if !ep && takes && board.piece_on(m.get_dest()).is_none() {
```

Comment:
Maybe extract `board.piece_on(m.get_dest())` into var?

Reply:
I doubt it really matters in release, since no mutation happens the compiler will probably do that for us. But I changed it.